### PR TITLE
[PVR] Fix and refactor PVR internal 'playing' states after removal of…

### PR DESCRIPTION
**Pull request by ksooo**
… VideoPlayer's 'other stream hack'."

Hopefully the last fallout from #12550. Fixes Live TV OSD for playing "dynpath" channels and other glitches.

This has been runtime tested on macOS, latest Kodi master.

@FernetMenta as discussed on Slack...

### CODE REVIEW

File: xbmc/addons/PVRClient.h [hunk 1]
```diff
@@ -723,9 +723,41 @@ namespace PVR
     bool IsPlayingEncryptedChannel(void) const;
     bool IsPlayingRecording(void) const;
     bool IsPlaying(void) const;
-    CPVRRecordingPtr GetPlayingRecording(void) const;
+
+    /*!
+     * @brief Set the channel that is currently playing.
+     * @param channel The channel that is currently playing.
+     */
+    void SetPlayingChannel(const CPVRChannelPtr channel);
```
Comment (FernetMenta) [inline xbmc/addons/PVRClient.h]: future PVR addons can play multiple channels at the same time. if an addon implementes inputstream, it knows the playing channels anyway. if it doesn't, I see not need for it to know.

Comment (ksooo) [inline xbmc/addons/PVRClient.h]: Sure, but for now, to make add-ons without an item demuxer working again, we need this. I'm aware that this code might go away soon, yes.

Comment (ksooo) [inline xbmc/addons/PVRClient.h]: My goal was to come up with a quick and (for what we have currently) not dirty fix.

Comment (ksooo) [inline xbmc/addons/PVRClient.h]: Without the other pieces for new pvr streaming being in place I'm unable to do this "right" nowadays.

File: xbmc/Application.cpp [hunk 2]
```diff
@@ -3501,6 +3501,8 @@ void CApplication::OnPlayBackEnded()
   CDarwinUtils::EnableOSScreenSaver(true);
 #endif
 
+  CServiceBroker::GetPVRManager().OnPlaybackEnded(m_itemCurrentFile);
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: the other direction that PVR updates currentFileItem in application should not be required anymore: https://github.com/xbmc/xbmc/search?utf8=%E2%9C%93&q=setcurrentfileitem&type=

File: xbmc/addons/PVRClient.h (hunk 1 above)
Comment (FernetMenta) [inline xbmc/addons/PVRClient.h]: ok

File: xbmc/Application.cpp (hunk 2 above)
Comment (ksooo) [inline xbmc/Application.cpp]: Absolutely, but not part/goal of this PR. This could be one of the next cleanup steps.

**Review by FernetMenta — APPROVED** [review]

File: xbmc/Application.cpp [hunk 3]
```diff
@@ -3568,6 +3572,8 @@ void CApplication::OnPlayBackStopped()
   CDarwinUtils::EnableOSScreenSaver(true);
 #endif
 
+  CServiceBroker::GetPVRManager().OnPlaybackStopped(m_itemCurrentFile);
+
```
Comment (da-anda) [inline xbmc/Application.cpp]: wouldn't it be cleaner to register PVRmanger and all the other components above to the respective announcement from CAnnouncementManager below? (same with onStart, etc). Would get rid of all the ifdeffery. Just a general question for some future refactoring that came to my mind when looking at this PRs changes.

Comment (da-anda) [inline xbmc/Application.cpp]: some other signal from cApplicationMessenger then?

Comment (FernetMenta) [inline xbmc/Application.cpp]: ApplicationMessenger is for messages in direction to the main thread.
This code is good as is. It is supposed to be snychronous, not to be hooked in to some undefined messaging anarchy

**Pull request merged**